src/Host.cpp

```diff
@@ -3212,17 +3212,24 @@ bool Host::echoWindow(const QString& name, const QString& text)
         return -1;
     }
 
-    auto pL = mpConsole->mLabelMap.value(name);
+    if (name.isEmpty() || name == QStringLiteral("main")) {
```

Comment:
This function is documented as obsolete - and it specifically does NOT target the main `TConsole` - so it shouldn't here either - the recommend replacement function - which you have also revised - did do something special for *that* `TConsole` so it is probably not wise to try and support it in this function...!

Reply:
Gaah. I totally missed that echo flag. I should get more sleep.

Will redo this.